h2o-core/src/main/java/water/ExternalFrameWriterClient.java

```diff
@@ -124,73 +124,74 @@ public void finalizeFrame(String keyName, long[] rowsPerChunk, byte[] colTypes,
     waitForRequestToFinish(timeout, requestType);
   }
 
-    public void close() throws IOException {
+    public void close() throws IOException, ExternalFrameConfirmationException {
+        waitForRequestToFinish(timeout, ExternalBackendRequestType.WRITE_TO_CHUNK.getByte());
```

Comment:
always wait for this request on backend to finish in close method as it was designed for it. In case the user of this API sends the empty partition no sent method will be called and therefore we never wait for request to finish